scheduler/server/job_state.go

```diff
@@ -197,3 +234,43 @@ func (j *jobState) getJobStatus() domain.Status {
 	}
 	return domain.InProgress
 }
+
+// addTaskToStartTimeMap add the running task to the map that bins running tasks by their start time
+func (j *jobState) addTaskToStartTimeMap(jobClass string, task *taskState, startTimeSec time.Time) {
+	if j.tasksByJobClassAndStartTimeSec == nil {
+		return
+	}
+	if _, ok := j.tasksByJobClassAndStartTimeSec[jobClass]; !ok {
+		j.tasksByJobClassAndStartTimeSec[jobClass] = taskTimeBuckets{}
+	}
+	if _, ok := j.tasksByJobClassAndStartTimeSec[jobClass][startTimeSec]; !ok {
```

Comment:
Looking up on jobClass is unnecessary. if we do an early return when ok is true, we can then preserve the reference to both taskTimeBuckets and taskStateByTaskID.

Reply:
done